src/lxc/cmd/lxc_usernsexec.c

```diff
@@ -223,46 +223,37 @@ static int read_default_map(char *fnam, int which, char *user)
 			continue;
 
 		newmap = malloc(sizeof(*newmap));
-		if (!newmap) {
-			fclose(fin);
-			free(line);
-			return -1;
-		}
+		if (!newmap)
+			goto on_error;
 
 		ret = lxc_safe_ulong(p1 + 1, &newmap->hostid);
-		if (ret < 0) {
-			fclose(fin);
-			free(line);
-			return -1;
-		}
+		if (ret < 0)
+			goto on_error;
 
 		ret = lxc_safe_ulong(p2 + 1, &newmap->range);
-		if (ret < 0) {
-			fclose(fin);
-			free(line);
-			return -1;
-		}
+		if (ret < 0)
+			goto on_error;
 
 		newmap->nsid = 0;
 		newmap->idtype = which;
 
 		tmp = malloc(sizeof(*tmp));
-		if (!tmp) {
-			fclose(fin);
-			free(line);
-			free(newmap);
-			return -1;
-		}
+		if (!tmp)
+			goto on_error;
```

Comment:
This now needs to set `ret = -1;` as it's 0 from the `lxc_safe_ulong()` call succeeding above.

Reply:
ffs

Reply:
fixed